ee/server/service/appconfig.go

```diff
@@ -0,0 +1,23 @@
+package service
+
+import (
+	"context"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+)
+
+func (svc *Service) HostFeatures(ctx context.Context, host *fleet.Host) (*fleet.Features, error) {
+	if host.TeamID != nil {
+		features, err := svc.ds.TeamFeatures(ctx, *host.TeamID)
+		if err != nil {
+			return nil, err
+		}
+		return features, nil
```

Comment:
So if I have host users enabled in global, and in team features I have only software inventory enabled explicitly, would that override host users and set it to false?

I would've expected this to first get `appConfig.Features` and then decode over it whatever happens to be defined within the team features.

Reply:
Currently:

1. When a team config is created, we do as you mention: copy all global configs, and then override with whatever happens to be defined within the team features.
2. From now on, the team config of `features` is independent of the global config

I made it this way to accommodate this request in the [epic](https://github.com/fleetdm/fleet/issues/6838):

> The override behavior should remain the same. The team ignores the global config. There is no merging between the two.

do you think this matches the spec or did I misinterpret the meaning of that statement?

Reply:
Ah, yeah, it matches it. Thanks for the clarification! I wonder if we should add a comment here?

Reply:
sounds good, I will 👍

Reply:
Do we have a test that cements this logic to make sure we don't break it?

Reply:
thanks! we were indirectly testing this but I missed to add an specific test.

both the test and the comment with the clarification were added in 29beefd90